meilisearch-http/src/routes/search.rs

```diff
@@ -53,45 +54,75 @@ pub async fn search_with_url_query(ctx: Request<Data>) -> SResult<Response> {
         search_builder.limit(limit);
     }
 
-    if let Some(attributes_to_retrieve) = query.attributes_to_retrieve {
-        for attr in attributes_to_retrieve.split(',') {
-            search_builder.add_retrievable_field(attr.to_string());
+    let available_attributes = schema.displayed_name();
+    let mut restricted_attributes: HashSet<&str>;
+    match &query.attributes_to_retrieve {
+        Some(attributes_to_retrieve) => {
+            let attributes_to_retrieve: HashSet<&str> = attributes_to_retrieve.split(',').collect();
+            if attributes_to_retrieve.contains("*") {
+                restricted_attributes = available_attributes.clone();
+            } else {
+                restricted_attributes = HashSet::new();
+                for attr in attributes_to_retrieve {
+                    if available_attributes.contains(attr) {
+                        restricted_attributes.insert(attr);
+                        search_builder.add_retrievable_field(attr.to_string());
+                    } else {
+                        warn!("The attributes {:?} present in attributesToCrop parameter doesn't exist", attr);
+                    }
+                }
+            }
+        },
+        None => {
+            restricted_attributes = available_attributes.clone();
         }
     }
 
     if let Some(attributes_to_crop) = query.attributes_to_crop {
-        let crop_length = query.crop_length.unwrap_or(200);
-        if attributes_to_crop == "*" {
-            let attributes_to_crop = schema
-                .displayed_name()
-                .iter()
-                .map(|attr| (attr.to_string(), crop_length))
-                .collect();
-            search_builder.attributes_to_crop(attributes_to_crop);
-        } else {
-            let attributes_to_crop = attributes_to_crop
-                .split(',')
-                .map(|r| (r.to_string(), crop_length))
-                .collect();
-            search_builder.attributes_to_crop(attributes_to_crop);
+        let default_length: usize = query.crop_length.unwrap_or(200);
```

Comment:
```suggestion
        let default_length = query.crop_length.unwrap_or(200);
```

Rust already know the type of `query.crop_length`.